### DISCUSSION TITLE
Data Importer several bugs / not working

### DISCUSSION THREAD

**Discussion by Drakon74** [firefly-iii/firefly-iii#9310] [Firefly III]
### Support guidelines

- [x] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [x] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [x] ... there are no open or closed issues that are related to my problem
- [x] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

So i setup the data importer and a firefly 3 instance recently. Both are brand new. Linking up the two gave me a little trouble at first, but i had forgot to put in "**" for trusted reverse proxies.

Anyway, so i wanted to use the importer to import a CSV File from my bank (Deutsche Sparkasse). To be fair it was a large file with one year worth of transactions.

Once i figured out the values and started the actual import, i notice a lot of bugs and issues with the importer.

First of, in some transactions the BIC is invalid. Thats just sadly how i get it from my bank. Though it looks like the transactions are skipped?

![Image](https://github.com/user-attachments/assets/81a5167f-2f61-4f9a-8943-cc5eb7b75ee3)

It then tried to import "income" transactions from my employer, though it couldn't find the accounts from it? I thought it would create them itself like it does with all expense accounts?

![Image](https://github.com/user-attachments/assets/039983df-e70b-4294-a1a2-61efbf6b57b4)

Eventuell the entire web interface crashed and wouldn't refresh anymore. Though maybe that is just my reverse proxy being weird. Refreshing the page helped, this kept happening for future imports too.

![Image](https://github.com/user-attachments/assets/a7796719-1472-4301-b6d3-aeea4e745c9d)

Eventually it was through... and my account now was nearly 40,000€ in the minus. I mean, i didn't expected it to be 100% the same (since some initial value is missing after all), but this is just too far off xD.

My suspicion was that it just couldn't create the income accounts and then just completely skipped over my salary transaction. Though looking in Firefly 3 itself, it did create the account... just with 2 transactions as opposed to 12:

![Image](https://github.com/user-attachments/assets/738424a3-fb0a-4364-8b7c-a723a9767438)

So i thought, okay, maybe re-importing will fix it now, and import the missing income transactions. So i tried that.

It did add the income transactions now just fine, but sadly imported nearly all expense transactions again, only for a few it noticed it being a duplicate, despite the duplication check being set to "Content-based".

So my account after all is still 40.000€ in the minus :D 

I'm sorry, but like this the importer is very unusable :/ 

It shouldn't skip over empty / invalid BIC values, i mean, thats not too important after all. 
It should create income accounts just fine without issue.
And the duplication check should also work. I exported the config settings beforehand and used exactly 1:1 the same settings and mapping. So i don't understand why in most cases the duplication check is not working at all.

But maybe i do something terrible wrong and just missed a checkbox or something?

Here are also my settings i used:

![Image](https://github.com/user-attachments/assets/74df7792-d505-4bfe-a850-16f82cb3aed2)
![Image](https://github.com/user-attachments/assets/6fbff501-1d1a-4975-819a-291c12f86bcc)
![Image](https://github.com/user-attachments/assets/682447e3-956a-4f03-999b-cc8e4c4c2fcc)
![Image](https://github.com/user-attachments/assets/c7d9c87a-c801-49e2-80af-8560358be584)
![Image](https://github.com/user-attachments/assets/540ee253-911d-4ec2-8475-2a9b53beaea9)

### Debug information

Debug information generated at 2024-10-05 11:00:11 for Firefly III version **v6.1.21**.

<table><thead><tr><th colspan="2">System information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Firefly III</td><td>6.1.21 / v2.1.0 / 24 (exp. 24)</td></tr><tr><td>PHP version</td><td>8.3.12 (64bits) / apache2handler / Linux x86_64</td></tr><tr><td>BCscale</td><td>12</td></tr><tr><td>Error reporting</td><td>Display: Off, reporting: ALL errors</td></tr><tr><td>Max upload</td><td>67108864 (64 MB)</td></tr><tr><td>Database drivers</td><td><strong>*mysql*</strong>, pgsql, sqlite, </td></tr><tr><td>Docker build</td><td><span>#</span>1131, base <span>#</span>91</td></tr></tbody></table><table><thead><tr><th colspan="2">Firefly III information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Timezone</td><td>Europe/Berlin + Europe/Berlin</td></tr><tr><td>App environment</td><td>production, debug: false</td></tr><tr><td>Layout</td><td>v1</td></tr><tr><td>Logging</td><td>notice, stack / (empty)</td></tr><tr><td>Cache driver</td><td>file</td></tr><tr><td>Default language and locale</td><td>en_US + equal</td></tr><tr><td>Trusted proxies</td><td>**</td></tr><tr><td>Login provider & user guard</td><td>eloquent / web</td></tr><tr><td>Login headers</td><td>N/A + N/A</td></tr><tr><td>Stateful domains</td><td></td></tr><tr><td>Last cron job</td><td>never (never)</td></tr><tr><td>Mailer</td><td>log</td></tr></tbody></table><table><thead><tr><th colspan="2">User-specific information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>User</td><td><span>#</span>1 of 1</td></tr><tr><td>User flags</td><td></td></tr><tr><td>Session start</td><td>2024-10-01 00:00:00</td></tr><tr><td>Session end</td><td>2024-10-31 23:59:59</td></tr><tr><td>View range</td><td>1M</td></tr><tr><td>User language</td><td>de_DE</td></tr><tr><td>User locale</td><td>de_DE</td></tr><tr><td>Locale(s) supported</td><td>de_DE.utf8: :white_check_mark:<br>de_DE.UTF-8: :white_check_mark:<br></td></tr><tr><td>User agent</td><td>Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/17.6 Safari/605.1.15</td></tr></tbody></table>

Debug information generated at 2024-10-05 11:03:56 Europe/Berlin for Firefly III Data Importer version **1.5.6**.

<table><tr><th colspan="2">System information</th></tr><tr><th>Item</th><th>Value</th></tr><tr><td>Version</td><td>1.5.6</td></tr><tr><td>Build</td><td><span>#</span>664, base <span>#</span>90</td></tr><tr><td>System</td><td>PHP 8.3.11, Linux, apache2handler</td></tr></table><table><tr><th colspan="2">App information</th></tr><tr><th>Item</th><th>Value</th></tr><tr><td>Timezone</td><td>Europe/Berlin, Europe/Berlin</td></tr><tr><td>Environment</td><td>local</td></tr><tr><td>Debug mode</td><td>false, cache 'file'</td></tr><tr><td>Log level</td><td>debug, stack</td></tr><tr><td>Display errors</td><td>Off, ALL errors</td></tr><tr><td>BCscale</td><td>12</td></tr><tr><td>Trusted proxies</td><td>**</td></tr></table><table><tr><th colspan="2">User information</th></tr><tr><th>Item</th><th>Value</th></tr><tr><td>User agent</td><td>Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/17.6 Safari/605.1.15</td></tr></table>

**Comment by JC5** [comment]
Hey, thanks for opening an issue. Sorry to hear you ran into issues. Let's go over this one by one.

First of all: start small. I suggest you split up your import file so you can import small batches, ie. 5 transactions at a time. That way you don't have this huge array of issues when you import data, and you can fine-tune your import.

Like you noticed, invalid BICs are not skipped. If your import contains invalid BIC values, sorry about that. Either remove them from the import or talk to your bank about it. This will not change.

> It then tried to import "income" transactions from my employer, though it couldn't find the accounts from it?

The importer will create the opposing accounts as revenue accounts. It depends a little bit on your column configuration. I see no obvious problems, but the logs may show the details.

> Refreshing the page helped, this kept happening for future imports too.

There is a button "start over", did you try it?

>  And the duplication check should also work. I exported the config settings beforehand and used exactly 1:1 the same settings and mapping. So i don't understand why in most cases the duplication check is not working at all.

I don't understand it either, but I will not be picking this up as a bug just yet. Depending on the import process, new accounts have been made (and recognized by the data importer). This is also a matter of details from the logs.

I hope I can help you with the import.

**Comment by Drakon74** [comment]
> Either remove them from the import or talk to your bank about it. This will not change.

I guess i will remove the BIC import entirely then. Not a huge issue either, just thought i bring this up.

> There is a button "start over", did you try it?

No, since this would re-start the entire import process wouldn't it? Just refreshing the page helped to get things going again. But then again, maybe thats just my reverse proxy being weird. The import process seems to work anyway in the background, just the page / reverse proxy getting a hiccup. I don't mind it too much honestly.

In any case, i just now tried to import "only" the last 3 months of transactions. I know, it is definitely more than 5 (about 200), but still thought i try, only thing different now is ignoring the BIC column completely. Otherwise the exact same settings.

Now everything worked.

I am very confused as to why, but the import worked just fine, the income accounts were created just fine, no errors at all. Balance looked believable afterwards (again no 100% match, but thats to be expected). Only the weird error 504 appeared again, but again, could be my reverse proxy.

I then added a little "initial" value so the balance matches and tried the import again... duplicate check now worked on all 200 transactions and not one was being added again.

I am super confused. Either the large file of nearly 1000 transactions was really too much or i really did something terribly wrong. 

In any case, now it all seems to work, i am sorry if i have wasted your time. Will observe the behavior the next months, but for now it seems fine weirdly.

**Reply by JC5** [reply]
> The import process seems to work anyway in the background, just the page / reverse proxy getting a hiccup. I don't mind it too much honestly.

Alright keep me posted. It's true that it may continue in the background, that's PHP for you 😅 

> I am super confused. 

I am too! :D It's weird but happy that it works now. Keep me posted on your progress. It should not make too much of a difference, 200 vs 1000 although it's possible one weird line threw everything off. 

Thanks for opening an issue!

**Comment by Thijxx** [comment]
I have the same issue with the current version, same errors, even importing 3 lines fails. Asset account IBAN always crashes any import, no matter the content or source file, just using the column means it will fail.
Ghost duplicates for many lines, it is set to detect on content but has to ignore all but 1 column for it to mark it as duplicates so this seems to be broken.
The Restart button is also broken, the web interface flickers but nothing else happens. If you found out what was going wrong I'm interested :)

**Reply by JC5** [reply]
So, it seems you have a different issue. Could you please open a new discussion and add logs or screenshots?
